Update provider code for carina-core API changes (read_data_source required, create_normalizer non-optional)

## Problem

After carina-rs/carina PRs #1853 and #1854:
- `Provider::read_data_source` is now a required method (no default impl)
- `ProviderFactory::create_normalizer` returns `Box<dyn ProviderNormalizer>` instead of `Option<Box<dyn ProviderNormalizer>>`

The provider WASM build fails with:
- `error[E0046]: not all trait items implemented, missing: read_data_source`
- `error[E0053]: method create_normalizer has an incompatible type for trait`

## Fix

1. Add `read_data_source` implementation to the provider (delegate to `self.read(&resource.id, None)` or implement properly)
2. Update `create_normalizer` return type from `Option<Box<dyn ProviderNormalizer>>` to `Box<dyn ProviderNormalizer>`

--- a/carina-provider-awscc/src/lib.rs
+++ b/carina-provider-awscc/src/lib.rs
@@ -125,8 +125,8 @@ impl ProviderFactory for AwsccProviderFactory {
     fn create_normalizer(
         &self,
         _attributes: &HashMap<String, Value>,
-    ) -> BoxFuture<'_, Option<Box<dyn ProviderNormalizer>>> {
-        Box::pin(async { Some(Box::new(AwsccNormalizer) as Box<dyn ProviderNormalizer>) })
+    ) -> BoxFuture<'_, Box<dyn ProviderNormalizer>> {
+        Box::pin(async { Box::new(AwsccNormalizer) as Box<dyn ProviderNormalizer> })
     }
 
     fn schemas(&self) -> Vec<carina_core::schema::ResourceSchema> {
@@ -177,6 +177,10 @@ impl Provider for AwsccProvider {
             self.read_resource(&id.resource_type, &id.name, identifier.as_deref())
                 .await
         })
+    }
+
+    fn read_data_source(&self, resource: &Resource) -> BoxFuture<'_, ProviderResult<State>> {
+        self.read(&resource.id, None)
     }
 
     fn create(&self, resource: &Resource) -> BoxFuture<'_, ProviderResult<State>> {

--- a/carina-provider-awscc/src/main.rs
+++ b/carina-provider-awscc/src/main.rs
@@ -167,6 +167,13 @@ impl CarinaProvider for AwsccProcessProvider {
         }
     }
 
+    fn read_data_source(
+        &self,
+        resource: &proto::Resource,
+    ) -> Result<proto::State, proto::ProviderError> {
+        self.read(&resource.id, None)
+    }
+
     fn create(&self, resource: &proto::Resource) -> Result<proto::State, proto::ProviderError> {
         let core_resource = convert::proto_to_core_resource(resource);
         let result = self

